loaders/dojo/i18nEval.js

```diff
@@ -15,26 +15,43 @@
  * limitations under the License.
  */
 
- /*
-  * Evaluates a Dojo i18n resource bundle and returns the bundle object
-  */
+/*
+ * Evaluates a Dojo i18n resource bundle and returns the bundle object
+ */
 module.exports = function(bundle) {
 	var result, isAmd;
 	function define(arg1, arg2) {
 		isAmd = true;
+
 		if (!arg2) {
+			// overwriting language files may define an easy object
 			result = arg1;
-		} else {
-			if (arg1.length !== 0) {
-				throw new Error("define dependencies not supported in langauge files!");
-			}
+
+		} else if (arg1.length === 0) {
+
+			// without dependencies we can easily call the empty factory function
 			result = arg2(); // call factory function
+
+		} else if (arg1.length === 2 && arg1[0] === "require" && arg1[1] === "exports") {
+
+			// Special case for typescript generated i18n files
+			// tsc is adding virtual dependencies for require and exports
+			// we need the latter as our result
+			var exp = {};
+			var requireFun = function() { throw new Error("require() is not supported in language files"); };
+			arg2(requireFun, exp);
```

Comment:
Disregard the previous update to my comment on this code.  The correct way to deal with AMD module exports is to use the exports object only if the define result is undefined.

```suggestion
			result = arg2(requireFun, exp);
			if (typeof result === 'undefined') {
				result = exp;
			}
```